webrecorder/browsertrix
# feat: Collection thumbnails, start page, and public view updates

## Changes

- Allows user to choose collection replay home page and collection thumbnail (resolves https://github.com/webrecorder/browsertrix/issues/2182)
- Displays collection thumbnails on org dashboard and public page
- Enables downloading public collection (resolves https://github.com/webrecorder/browsertrix/issues/2233)
- Adds caption as "Summary" to metadata dialog
- Moves description editor to "About" tab

## Manual testing

(The manual testing steps are left deliberately vague to test the usability of this feature. Open to any UX related feedback.)

Backend must be deployed locally using this branch: https://github.com/webrecorder/browsertrix/pull/2198

Log in as crawler to public org. Verify the following:

- Update and view collection summary and description
- Enable and disable download button on unlisted/public collection page
- Change the collection's default thumbnail in "Share Settings"
- Choose a Replay home URL and optionally set the thumbnail to the URL preview
- Browse public collection starting from its home URL
- Public collections display correctly on smaller screen sizes

## Screenshots

| Page | Image/video |
| ---- | ----------- |
| Org Dashboard | **No public collections:**<br/><img width="766" alt="Screenshot 2024-12-17 at 11 24 52 AM" src="https://github.com/user-attachments/assets/655c4692-a331-4617-abb5-9bee5b3131d4" /> |
| Collections | **Action menu:**<br/><img width="994" alt="Screenshot 2024-12-17 at 11 41 11 AM" src="https://github.com/user-attachments/assets/7add4a03-a409-4fd3-9941-8e2eac4e5d41" /> |
| Collections | **New collection dialog:**<br/><img width="658" alt="Screenshot 2024-12-17 at 12 03 36 PM" src="https://github.com/user-attachments/assets/9eb2b5d8-c30d-4ed6-a320-7e4804667bb8" /> |
| Collection Detail | **Edit about:**<br/><img width="1017" alt="Screenshot 2024-12-17 at 11 33 49 AM" src="https://github.com/user-attachments/assets/a7c5e2f8-8f0f-4fdb-a3f2-6cb84433d177" /> |
| Collection Detail | **Action. menu:**<br/><img width="305" alt="Screenshot 2024-12-17 at 11 34 33 AM" src="https://github.com/user-attachments/assets/bbea2031-50b4-4297-9690-b2413a6fe499" /> |
| Collection Detail | **Configure replay home dialog:**<br/><img width="979" alt="Screenshot 2024-12-17 at 11 34 49 AM" src="https://github.com/user-attachments/assets/698a47fb-76d2-48eb-875c-64867f20cbbc" /><br/><br/><img width="975" alt="Screenshot 2024-12-17 at 11 34 57 AM" src="https://github.com/user-attachments/assets/e96e13f6-3af0-4020-ba3d-3b64e1e391a6" /><br/><br/><img width="981" alt="Screenshot 2024-12-17 at 11 35 14 AM" src="https://github.com/user-attachments/assets/f4733ed9-11c0-4457-bb08-700d20e0e6cf" /> |
| Collection Detail | **Choose thumbnail:**<br/><img width="652" alt="Screenshot 2024-12-17 at 4 28 54 PM" src="https://github.com/user-attachments/assets/b728a1a6-a399-4a27-babf-1173daebf794" /> |
| Dashboard | **With public collections:**<br/><img width="1324" alt="Screenshot 2024-12-17 at 11 46 54 AM" src="https://github.com/user-attachments/assets/943ad7e3-0afa-4051-ac0d-558f9841c4c4" /> |
| Public Org Profile | <img width="1425" alt="Screenshot 2024-12-17 at 11 54 00 AM" src="https://github.com/user-attachments/assets/2bbba7a3-8651-49ce-bfac-858dcc80649b" /> |
| Public Collection | **Replay with home URL:**<br/><img width="969" alt="Screenshot 2024-12-17 at 11 35 41 AM" src="https://github.com/user-attachments/assets/5f4c7008-4a79-4739-9c13-d36c6631d2e1" /> |
| Public Collection | **About:**<br/><img width="964" alt="Screenshot 2024-12-17 at 11 35 56 AM" src="https://github.com/user-attachments/assets/6a2f7e99-7764-4c88-a064-94935031d69b" /> |
| Public Collection | **No description provided:**<br/><img width="984" alt="Screenshot 2024-12-17 at 11 33 33 AM" src="https://github.com/user-attachments/assets/8cc3e422-7f23-4d29-b9d4-cb02f8bf4043" /> |
